public/components/panel/studioPanel/sampleDownloader.js

```diff
@@ -54,11 +68,13 @@ class SampleDownloader extends React.Component {
       .then(audioBuffer => {
         if (SampleDownloaderMounted) {
           sample.sampleAudioBuffer = audioBuffer
-          sample.sampleAudioLength = audioBuffer.duration
-          this.props.onComplete(sample)
+          sample.sampleAudioLength = sample.sampleAudioLength === -1 ? audioBuffer.duration : -1
+          prevResults.push(sample)
+          this.startDownload(prevResults, downloadIndex + 1)
```

Comment:
This uses recursion here, which usually is a no-no buttttttt.. there's a hardcoded maximum barrier of a 100 samples in a beat right now so stack-overflow isn't a problem. **TL/DR**: There can be at-most 100 stacks.